Add Window-compatible start script generation

jchab opened the pull request:
If Windows is detected, a Windows-compatible .bat script is
generated. If a different OS is detected, the traditional bash
script is generated.

Commit:
Create Windows-compatible start script

If Windows is detected, a Windows-compatible .bat script is
generated. If a different OS is detected, the traditional bash
script is generated.

havocp commented on src/main/scala/com/typesafe/sbt/SbtStartScript.scala:
```diff
@@ -251,24 +259,36 @@ object SbtStartScript extends Plugin {
+                if (isWindows()) {
+                    """set MAINCLASS="$1"
+SHIFT
+if "%MAINCLASS%"=="" ( echo 'This "start" script requires a main class name as the first argument, because a mainClass was not specified in SBT and not autodetected by SBT (usually means you have zero, or more than one, main classes).  You could specify in your SBT build: mainClass in Compile := Some("Whatever")' EXIT 1)
```
It might be nice to factor out this error message so it isn't pasted in two places

jchab replied:
Good call. I addressed this in cb9a733

havocp commented on src/main/scala/com/typesafe/sbt/SbtStartScript.scala:
```diff
@@ -369,11 +415,16 @@ exec java $JAVA_OPTS -Djetty.port="$PORT" -Djetty.home="@JETTY_HOME@" -jar "@JET
     def startScriptNotDefinedTask(streams: TaskStreams, scriptFile: File) = {
-        writeScript(scriptFile, """#!/bin/bash
+        val msgWindows = """
+echo "No meaningful way to start this project was defined in the SBT build" 1>&2
```
similarly here, just one copy of the error message possible?

jchab replied:
See cb9a733

havocp commented:
do the 'scripted' tests now pass on Windows? (see http://eed3si9n.com/testing-sbt-plugins )

havocp commented:
Thanks!

Commit:
Fix a few issues with Windows script creation

There were a few linux-isms left in the Windows script
generation that showed themselves when the test suite was run.
Consolidated error strings into variables.

Commit:
Fix tests to reference latest plugin version

jchab commented:
The tests required updates to the version numbers in plugins.sbt (a43bc1d). I was able to successfully run all three tests in Windows (with a few small tweaks to my code in cb9a733).

havocp commented:
Thanks! One more hoop I forgot about - can you click through http://www.typesafe.com/contribute/cla with your github account? It should just be a couple of clicks to register @jchab as a contributor. Then this looks good to go to me.

jchab commented:
Done! Thanks!

havocp commented:
Looks good, appreciate the patch.

jchab commented:
Not a problem. Thanks for the merge! Would it be at all possible for you to cut a release with my changes?

havocp commented:
v0.8.0 cut. I think anyway. Not sure if it's live yet.

jchab commented:
Thanks!!
